node/src/types/block.rs

```diff
@@ -1204,17 +1207,30 @@ pub struct FinalitySignature {
 
 impl FinalitySignature {
     /// Create an instance of `FinalitySignature`.
-    pub fn new(block_hash: BlockHash, signature: Signature, public_key: PublicKey) -> Self {
+    pub fn new(
+        block_hash: BlockHash,
+        era_id: EraId,
+        secret_key: &Rc<SecretKey>,
```

Comment:
It's less restrictive to just require `&SecretKey` here. If you have an `sk: Rc<SecretKey>`, you can always pass in `&*sk`.

Reply:
Done in [`c0bd13f` (#683)](https://github.com/CasperLabs/casper-node/pull/683/commits/c0bd13f70e41208382667535631dd62c9a381c5c)